squirrel.frontier/src/main/java/org/dice_research/squirrel/frontier/impl/FrontierQueryGenerator.java

```diff
@@ -0,0 +1,212 @@
+package org.dice_research.squirrel.frontier.impl;
+import java.util.Collection;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import org.apache.jena.graph.Node;
+import org.apache.jena.graph.Triple;
+import org.apache.jena.query.Query;
+import org.apache.jena.query.QueryFactory;
+
+public class FrontierQueryGenerator {
+	   /**
+     * The instance of the class QueryGenerator.
+     */
+    private static final FrontierQueryGenerator instance = new FrontierQueryGenerator();
+    String PREFIX= "PREFIX xsd: <http://www.w3.org/2001/XMLSchema#>";
+    @SuppressWarnings("unused")
+    private static final Logger LOGGER = LoggerFactory.getLogger(FrontierQueryGenerator.class);
+
+    private FrontierQueryGenerator() {
+    }
+
+    /**
+     * Getter for {@link #instance}.
+     *
+     * @return instannce of the class.
+     */
+    public static FrontierQueryGenerator getInstance() {
+        return instance;
+    }
+
+    /**
+     * Return an Add Query for the default uri and its triples.
+     *
+     * @param listBufferedTriples the given list of triples.
+     * @return The generated query.
+     */
+    public String getAddQuery(Collection<Triple> listBufferedTriples) {
+        return getAddQuery(null, listBufferedTriples, true);
+    }
+
+    /**
+     * Return an Add Query for the given uri and its triples.
+     *
+     * @param graphId             the graph id where the triples are stored.
+     * @param listBufferedTriples the given list of triples.
+     * @return The generated query.
+     */
+    public String getAddQuery(String graphId, Collection<Triple> listBufferedTriples) {
+        return getAddQuery(graphId, listBufferedTriples, false);
+    }
+
+    /**
+     * Return an Add Query for the given uri or default graph and its triples.
+     *
+     * @param graphId                 the graph id where the triples are stored.
+     * @param listBufferedTriples the given list of triples.
+     * @param defaultGraph Identify if query is for the default graph.
+     * @return The generated query.
+     */
+    public String getAddQuery(String graphId, Collection<Triple> listBufferedTriples, boolean defaultGraph) {
+        StringBuilder stringBuilder = new StringBuilder();
```

Comment:
It might be much easier to create an INSERT query using jena. Please take a look at https://github.com/hobbit-project/core/blob/master/src/main/java/org/hobbit/storage/queries/SparqlQueries.java#L871-L893
Maybe you can even reuse the method from there (without copying it).